portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -99,29 +100,22 @@ private int getRequestedCommentCount(HttpServletRequest request) {
     return commentCount;
   }
 
-  public int getIdNum() {
-    int idNum = 0;
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      String idString = (String) entity.getProperty("id");
-      int id = Integer.parseInt(idString);
-      if (id > idNum) {
-        idNum = id;
-      }
-    }
-    idNum++;
-    return idNum;
-  }
+  private int idNum = -1;
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    int idNum = getIdNum();
+
     String userComment = request.getParameter("comment");
     String userName = request.getParameter("name");
     long timestamp = System.currentTimeMillis();
 
+    if (idNum == -1) {
+      ServletContext sc = request.getServletContext();
+	  String id = sc.getAttribute("id").toString();
```

Comment:
I'd recommend you use a more specific name for the attribute here, something like `"nextAvailableCommentId"` to signal its intended use. I would also advice you use a static constant to store this string rather than hard-code the string in two different places.